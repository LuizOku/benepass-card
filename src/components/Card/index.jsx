import React, { useMemo } from "react";
import Logo from "../../assets/images/Logo.svg";
import Visa from "../../assets/images/Visa.svg";
import { formatCardNumber } from "../../utils/format";

function Card({ number, type, validThru, cvc, zip, shouldShowDetails }) {
  const memoizedNumber = useMemo(() => {
    if (shouldShowDetails) {
      return formatCardNumber(number);
    }
    return `●●●● ${number.slice(-4)}`;
  }, [shouldShowDetails, number]);

  const memoizedCVC = useMemo(() => {
    return shouldShowDetails ? cvc : "●●●";
  }, [shouldShowDetails, cvc]);

  return (
    <div
      data-testid="card-component"
      className="bg-card bg-cover rounded-xl p-6 sm:w-[392px] h-[272px] min-w-[310px] flex flex-col transition-all duration-500 md:hover:scale-105"
    >
      <div className="w-full flex items-center justify-between mb-6">
        <img
          data-testid="card-logo"
          src={Logo}
          alt="Logo"
          className="w-[121px] h-8 object-cover"
        />
        <span className="px-3 py-1 rounded-full bg-white capitalize text-base font-medium text-brand-indigo-100">
          {type}
        </span>
      </div>
      <span className="text-[26px] leading-8 text-white">{memoizedNumber}</span>
      <div className="mt-6 flex flex-row items-center justify-between w-1/2">
        <div className="flex flex-col">
          <span className="text-xs font-medium text-white">VALID THRU</span>
          <span className="text-[18px] leading-6 font-medium text-white">
            {validThru}
          </span>
        </div>
        <div className="flex flex-col">
          <span className="text-xs font-medium text-white">CVC</span>
          <span className="text-[18px] leading-6 font-medium text-white">
            {memoizedCVC}
          </span>
        </div>
      </div>
      <div className="flex-grow"></div>

      <div className="w-full flex items-center justify-between">
        <span className="text-[18px] leading-6 font-normal text-white">
          {`ZIP ${zip}`}
        </span>
        <img
          data-testid="visa-img"
          src={Visa}
          alt="Visa"
          className="w-[54px] h-[18px] object-cover"
        />
      </div>
    </div>
  );
}

export default Card;
