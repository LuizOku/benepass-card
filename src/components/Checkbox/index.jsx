import React from "react";

function Checkbox({ label, value, onChange, className }) {
  return (
    <div
      data-testid="checkbox-component"
      className={`flex items-center ${className}`}
    >
      <input
        data-testid="checkbox"
        type="checkbox"
        checked={value}
        onChange={onChange}
        className="mr-3 transform scale-110 border-0 outline accent-brand-pink-100 cursor-pointer transition-all duration-500 md:hover:scale-125"
      />
      <label
        className={`text-base font-normal ${value && "text-brand-pink-100"}`}
      >
        {label}
      </label>
    </div>
  );
}

export default Checkbox;
