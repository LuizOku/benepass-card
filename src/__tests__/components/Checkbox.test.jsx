import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { Checkbox } from "../../components";

const baseProps = {
  label: "Show details",
  onChange: jest.fn(),
  value: true,
};

const renderComponent = (customProps) => {
  const props = { ...baseProps, ...customProps };

  return render(<Checkbox {...props} />);
};

describe("<Checkbox />", () => {
  it("renders without crashing", () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it("should render the label", () => {
    renderComponent();
    expect(screen.getByText(baseProps.label)).toBeInTheDocument();
  });

  it("should call onChange handler when checkbox is clicked", () => {
    const onChangeMock = jest.fn();
    renderComponent({ onChange: onChangeMock });

    fireEvent.click(screen.getByTestId("checkbox"));

    expect(onChangeMock).toHaveBeenCalledTimes(1);
  });
});
