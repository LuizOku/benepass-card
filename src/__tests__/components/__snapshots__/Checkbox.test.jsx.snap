// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center undefined"
        data-testid="checkbox-component"
      >
        <input
          checked=""
          class="mr-3 transform scale-110 border-0 outline accent-brand-pink-100 cursor-pointer transition-all duration-500 md:hover:scale-125"
          data-testid="checkbox"
          type="checkbox"
        />
        <label
          class="text-base font-normal text-brand-pink-100"
        >
          Show details
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center undefined"
      data-testid="checkbox-component"
    >
      <input
        checked=""
        class="mr-3 transform scale-110 border-0 outline accent-brand-pink-100 cursor-pointer transition-all duration-500 md:hover:scale-125"
        data-testid="checkbox"
        type="checkbox"
      />
      <label
        class="text-base font-normal text-brand-pink-100"
      >
        Show details
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
